admin-dev/themes/new-theme/js/app/pages/permission/components/row.vue

```diff
@@ -0,0 +1,324 @@
+     *
+     * @author    PrestaShop SA <contact@prestashop.com>
+     * @copyright 2007-2019 PrestaShop SA and Contributors
+     * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+     * International Registered Trademark & Property of PrestaShop SA
+     *-->
+<template>
+  <div>
+    <div
+      :class="{parent, 'bg-light': parent}"
+      class="row permission-row"
+    >
+      <div class="col-4">
+        <span v-html="displayLevelDepth" />
+        &raquo;
+
+        <strong v-if="parent">{{ permission.name }}</strong>
+        <template v-else>
+          {{ permission.name }}
+        </template>
+      </div>
+
+      <div class="col-8 row">
+        <div
+          class="text-center"
+          :class="getClasses(types, index === 0)"
+          v-for="(type, index) in types"
+          :key="index"
+        >
+          <ps-checkbox
+            :value="type"
+            v-model="permissionValues"
+            @change="sendUpdatePermissionRequest(type)"
+            :disabled="!canEdit || !canEditCheckbox(type)"
+          />
+        </div>
+      </div>
+    </div>
+
+    <div v-if="permission.children !== undefined">
+      <row
+        v-for="(p, pId) in permission.children"
+        :key="p.id"
+        :can-edit="canEdit"
+        :permission="p"
+        :permission-id="pId"
+        :permission-key="permissionKey"
+        :level-depth="levelDepth + 1"
+        :profile-permissions.sync="profilePermissions"
+        :employee-permissions="employeePermissions"
+        :types="types"
+        @childUpdated="onChildUpdate"
+        @sendRequest="sendRequest"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+  import PsCheckbox from '@app/components/checkbox.vue';
+  import ColSize from '@app/mixins/col-size.vue';
+
+  export default {
+    name: 'Row',
+    mixins: [
+      ColSize,
+    ],
+    components: {
+      PsCheckbox,
+    },
+    props: {
+      parent: {
+        type: Boolean,
+        required: false,
+        default: false,
+      },
+      profilePermissions: {
+        type: Object,
+        required: true,
+      },
+      employeePermissions: {
+        type: Object,
+        required: false,
+        default: () => ({}),
+      },
+      permission: {
+        type: Object,
+        required: true,
+      },
+      permissionId: {
+        type: String,
+        required: true,
+      },
+      permissionKey: {
+        type: String,
+        required: true,
+      },
+      levelDepth: {
+        type: Number,
+        required: true,
+      },
+      canEdit: {
+        type: Boolean,
+        required: false,
+        default: false,
+      },
+      types: {
+        type: Array,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        permissionValues: [],
+        TYPE_ALL: 'all',
+      };
+    },
+    watch: {
+      profilePermissions: {
+        handler: function mandatoryFunctionForDeepWatching() {
```

Comment:
just a question. why can't we use notation
mandatoryFunctionForDeepWatching = () => {
   ...
}

Reply:
Because of the `this` statement, otherwise, it's not interpreted